docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -408,26 +566,29 @@ Not handling the error cases correctly is risky: for example, if a ledger transf
 
 When making inter-canister calls, always handle the error cases (rejects) correctly. These errors imply that the message has not been successfully executed.
 
-### Only make inter-canister calls to trustworthy canisters
+### Be aware of the risks involved in calling untrustworthy canisters
 
 #### Security concern
 
 - If inter-canister calls are made to potentially malicious canisters, this can lead to DoS issues or there could be issues related to candid decoding. Also, the data returned from a canister call could be assumed to be trustworthy when it is not.
 
-- If a canister is called with a callback, the receiver can stall indefinitely if the peer does not respond, resulting in DoS. A canister can no longer be upgraded if it is in that state. Recovery would involve reinstalling, wiping the state of the canister.
+- If a canister is called with a callback, the receiver can stall indefinitely if the peer does not respond, resulting in DoS. A canister can no longer be upgraded if it is in that state. Recovery would involve reinstalling, wiping the state of the canister. Note that even a trustworthy canister could have a bug causing it to stall indefinitely. However, such a bug seems rather unlikely to occur. 
 
 - In summary, this can DoS a canister, consume an excessive amount of resources, or lead to logic bugs if the behavior of the canister depends on the inter-canister call response.
 
 #### Recommendation
 
-- Only make inter-canister calls to trustworthy canisters.
+- Making inter-canister calls to trustworthy canisters is safe, except for the rather unlikely case that there is a bug in the callee that makes it stall forever. 
```

Comment:
Actually, you also need to trust the target subnet. This is relevant for canisters on the NNS but might also apply in the future where we have rented subnets or subnets with smaller replication factors.

Reply:
I think trusting a canister implies trusting its subnet and various other dependencies. I'm not aware if we have documentation about what trusting a canister entails, but it if not it would be useful to create it and link to it from here.

Reply:
Added a #2818. Will handle the issue outside of this PR.